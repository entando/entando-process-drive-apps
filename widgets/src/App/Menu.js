import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import {
  Attachment as AttachmentIcon,
  Home as HomeIcon,
  ViewQuilt as ViewQuiltIcon,
  List as ListIcon,
  BarChart as BarChartIcon,
  Settings as SettingsIcon,
  Dashboard as DashboardIcon,
  Description as DescriptionIcon,
  DoneAll as DoneAllIcon,
  Comment as CommentsIcon,
  CloudQueue as CloudQueueIcon,
  DeveloperBoard as DeveloperBoardIcon,
} from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  item: {
    position: 'relative',
  },
  listItem: {
    paddingLeft: 10,
    paddingRight: 10,
  },
  iconButton: {
    position: 'absolute',
    right: 5,
    top: 12,
  },
};

const menuItems = [
  {
    linkTo: '/Connections',
    linkText: 'Connections',
    Icon: <CloudQueueIcon />,
  },
  {
    linkTo: '/TaskList',
    linkText: 'Task list',
    Icon: <ListIcon />,
    configLinkTo: '/TaskListConfig',
  },
  {
    linkTo: '/TaskDetails',
    linkText: 'Task details',
    Icon: <DescriptionIcon />,
    configLinkTo: '/TaskDetailsConfig',
  },
  {
    linkTo: '/TaskCompletionForm',
    linkText: 'Task completion form',
    Icon: <DoneAllIcon />,
    configLinkTo: '/TaskCompletionFormConfig',
  },
  {
    linkTo: '/TaskComments',
    linkText: 'Task comments',
    Icon: <CommentsIcon />,
    configLinkTo: '/TaskCommentsConfig',
  },
  {
    linkTo: '/SummaryCard',
    linkText: 'Summary card',
    Icon: <DashboardIcon />,
    configLinkTo: '/SummaryCardConfig',
  },
  {
    linkTo: '/ProcessForm',
    linkText: 'Process form',
    Icon: <DescriptionIcon />,
    configLinkTo: '/ProcessFormConfig',
  },
  {
    linkTo: '/ProcessDefinition',
    linkText: 'Process Definition',
    Icon: <DeveloperBoardIcon />,
    configLinkTo: 'ProcessDefinitionConfig',
  },
  {
    linkTo: '/OvertimeGraph',
    linkText: 'Overtime graph',
    Icon: <BarChartIcon />,
    configLinkTo: '/OvertimeGraphConfig',
  },
  {
    linkTo: '/Attachments',
    linkText: 'Attachments',
    Icon: <AttachmentIcon />,
    configLinkTo: '/AttachmentsConfig',
  },
  {
    linkTo: '/ProcessList',
    linkText: 'Process List',
    Icon: <ListIcon />,
    configLinkTo: '/ProcessListConfig',
  },
];

const Menu = ({ open, setOpen, classes }) => (
  <Drawer open={open} onClose={() => setOpen(false)}>
    <div className="app-list">
      <List>
        <ListItem button component={Link} to="/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <HomeIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Home" />
        </ListItem>

        <ListItem button component={Link} to="/task-details-page/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <ViewQuiltIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Task Details page" />
        </ListItem>

        <ListItem button component={Link} to="/smart-inbox-page/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <ViewQuiltIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Smart Inbox page" />
        </ListItem>

        <Typography variant="overline">Widgets</Typography>
        <Divider />
        {menuItems.map(({ linkTo, Icon, linkText, configLinkTo }) => (
          <div className={classes.item} key={linkTo}>
            <ListItem
              className={classes.listItem}
              button
              component={Link}
              to={linkTo}
              onClick={() => setOpen(false)}
            >
              {Icon}
              <ListItemText style={{ marginLeft: 5 }} primary={linkText} />
            </ListItem>
            {configLinkTo && (
              <IconButton
                className={classes.iconButton}
                size="small"
                component={Link}
                to={configLinkTo}
                onClick={() => setOpen(false)}
              >
                <SettingsIcon fontSize="small" />
              </IconButton>
            )}
          </div>
        ))}
      </List>
    </div>
  </Drawer>
);

Menu.propTypes = {
  classes: PropTypes.shape({
    item: PropTypes.string,
    listItem: PropTypes.string,
    iconButton: PropTypes.string,
  }),
  open: PropTypes.bool,
  setOpen: PropTypes.func,
};

Menu.defaultProps = {
  classes: {},
  open: false,
  setOpen: () => {},
};

export default withStyles(styles)(Menu);
