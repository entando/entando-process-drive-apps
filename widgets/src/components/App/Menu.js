import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Typography,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  Drawer,
  Divider,
  IconButton,
} from '@material-ui/core';
import {
  Home as HomeIcon,
  List as ListIcon,
<<<<<<< HEAD
  Settings as SettingsIcon,
  Dashboard as DashboardIcon,
  Description as DescriptionIcon,
  DoneAll as DoneAllIcon,
  Comment as CommentsIcon,
=======
  BarChart as BarChartIcon,
  Settings as SettingsIcon,
>>>>>>> PDA-71 Fix styles
} from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  item: {
    position: 'relative',
  },
  iconButton: {
    position: 'absolute',
    right: 5,
    top: 11,
  },
};

const menuItems = [
  {
    linkTo: '/TaskList',
    linkText: 'Task list',
    Icon: <ListIcon />,
    configLinkTo: '/TaskListConfig',
  },
  {
    linkTo: '/TaskDetails',
    linkText: 'Task details',
    Icon: <DescriptionIcon />,
    configLinkTo: '/TaskDetailsConfig',
  },
  {
    linkTo: '/TaskCompletionForm',
    linkText: 'Task completion form',
    Icon: <DoneAllIcon />,
    configLinkTo: '/TaskCompletionFormConfig',
  },
  {
    linkTo: '/TaskComments',
    linkText: 'Task comments',
    Icon: <CommentsIcon />,
    configLinkTo: '/TaskCommentsConfig',
  },
  {
    linkTo: '/SummaryCard',
    linkText: 'Summary card',
    Icon: <DashboardIcon />,
    configLinkTo: '/SummaryCardConfig',
  },
  {
    linkTo: '/ProcessForm',
    linkText: 'Process form',
    Icon: <DescriptionIcon />,
    configLinkTo: '/ProcessFormConfig',
  },
  {
    linkTo: '/OvertimeGraph',
    linkText: 'Overtime graph',
    Icon: <BarChartIcon />,
    configLinkTo: '/OvertimeGraphConfig',
  },
];

const Menu = ({ open, setOpen, classes }) => (
  <Drawer open={open} onClose={() => setOpen(false)}>
    <div className="app-list">
      <List>
        <ListItem button component={Link} to="/" onClick={() => setOpen(false)}>
          <ListItemIcon>
            <HomeIcon />
          </ListItemIcon>
          <ListItemText primary="Home" />
        </ListItem>

        <Typography variant="overline">Widgets</Typography>
        <Divider />
        {menuItems.map(({ linkTo, Icon, linkText, configLinkTo }) => (
          <div className={classes.item} key={linkTo}>
            <ListItem button component={Link} to={linkTo} onClick={() => setOpen(false)}>
              {Icon}
              <ListItemText primary={linkText} />
            </ListItem>
            <IconButton
              className={classes.iconButton}
              size="small"
              component={Link}
              to={configLinkTo}
              onClick={() => setOpen(false)}
            >
              <SettingsIcon fontSize="small" />
            </IconButton>
          </div>
        ))}
      </List>
    </div>
  </Drawer>
);

Menu.propTypes = {
  classes: PropTypes.shape({
    item: PropTypes.string,
    iconButton: PropTypes.string,
  }),
  open: PropTypes.bool.isRequired,
  setOpen: PropTypes.func.isRequired,
};

Menu.defaultProps = {
  classes: {},
};

export default withStyles(styles)(Menu);
