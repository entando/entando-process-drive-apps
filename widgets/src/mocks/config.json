{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
