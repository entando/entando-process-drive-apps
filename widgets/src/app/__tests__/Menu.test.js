import { render } from '@testing-library/react';
import React from 'react';

import 'mocks/i18nMock';

import Menu from 'app/Menu';

describe('<Menu />', () => {
  it('renders snapshot correctly', () => {
    const { container } = render(<Menu />);

    expect(container).toMatchSnapshot();
  });
});
