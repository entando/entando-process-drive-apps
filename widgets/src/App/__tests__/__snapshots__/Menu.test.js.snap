// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders snapshot correctly 1`] = `<div />`;
